src/common/pf_eth.c

```diff
@@ -69,27 +109,23 @@ int pf_eth_init (pnet_t * net, const pnet_cfg_t * p_cfg)
    while (port != 0)
    {
       p_port_data = pf_port_get_state (net, port);
-      p_port_cfg = pf_port_get_config (net, port);
 
 #if PNET_MAX_PORT > 1
-      p_port_data->eth_handle = pnal_eth_init (
-         p_port_cfg->phy_port.if_name,
-         PNAL_ETHTYPE_LLDP,
-         pf_eth_recv,
-         (void *)net);
-#else
-      /* In single port configuration managed port is also physical port 1 */
-      p_port_data->eth_handle = net->eth_handle;
-#endif
+      p_port_cfg = pf_port_get_config (net, port);
 
-      if (p_port_data->eth_handle == NULL)
+      if (
+         pf_eth_init_netif (
+            net,
+            p_port_cfg->port_netif.name,
+            PNAL_ETHTYPE_LLDP,
+            &p_port_data->netif) != 0)
       {
-         LOG_ERROR (
-            PNET_LOG,
-            "Failed to init \"%s\"\n",
-            p_port_cfg->phy_port.if_name);
          return -1;
       }
+#else
+      /* In single port configuration managed port is also physical port 1 */
```

Comment:
Maybe
`/* In single port configuration is managed port also physical port 1 */`

Reply:
Rephrased, done